# 1. Environment setup

#### **Install softwares in your home directory**

Login in to OSC through:```ssh -X osu10343@owens.osc.edu```

Go to home directory:```cd $HOME ```

Make sure you are in home directory: ```pwd ```

Load R and start: ```module load R/3.5.0```

Run R: ```R```

Add Bioconductor and then the Rsubread , Rbowtie2, and edgeR package:

```
source("https://bioconductor.org/biocLite.R")
biocLite()
biocLite("Rsubread")
biocLite("Rbowtie2")
biocLite("edgeR")
quit() n
```
Check the softwares are installed by checking directory: ```ls ./R/x86_64-pc-linux-gnu-library/3.5/```

Make sure you have BiocInstaller, edgeR, limma, locfit, Rbowtie2, and Rsubread subdirectories in the folder.

#### **Set up directories in your scratch space**

```
cd /fs/scratch/PAS1407/osu10343/
mkdir RNAseq
cd RNAseq
mkdir RiceSeq Raw_Data Rcodes Rsubread Rbowtie2
```

Copy reference genome

```
cd RiceSeq
cp /fs/scratch/PAS1407/HCS7194_Files/Variant_Calling/all.dir/all.chrs.con .
cp /fs/scratch/PAS1407/HCS7194_Files/Variant_Calling/all.dir/all.gff3 .
```

Check the files using ```head -n 10 all.chrs.con``` and ```head -n 10 all.gff3``` 

Notice how the chromosome is listed >Chr1 and gene name is given as Name=LOC_Os##g#####.#

Download Raw_Data and code

```
cd ../Raw_Data/
cp /fs/scratch/PAS1407/HCS7194_Files/Gene_Expression/Raw_Data/*.* .
```

Copy Rcodes

```
cd ../Rcodes/
cp /fs/scratch/PAS1407/HCS7194_Files/Gene_Expression/Rcodes/*.* .
```

#### **Set up R script**

Go to Home:```cd ~```

Copy R code to Home Directory:```cp /fs/scratch/PAS1407/osu10343/RNAseq/Rcodes/*.* .```

Open the R script for the index builder and make modification (changing the directory name):```nano SubreadIndex.R```
In Nano: use ALT+R to replace XXXXX to your account number, use Ctrl+W to search

#### **Creat Bash file and submit to OSC**

```
echo "#PBS -N Rsubread_Index_Rice
#PBS -l walltime=02:00:00
#PBS -l nodes=1:ppn=28,mem=110GB
#PBS -j oe
#PBS -m bea
#PBS -A PAS1407
cd $HOME
module load R/3.5.0
R CMD BATCH --quiet --no-restore --no-save ./SubreadIndex.R >SubreadIndex.OUT#" > script
```

then ```qsub script```

Open the R script for Aligner and make modification (changing the directory name):```nano SubreadAlign.R```

Creat Bash file and submit to OSC

```
echo "#PBS -N Rsubread_Rice_align_RNAseq
#PBS -l walltime=02:00:00
#PBS -l nodes=1:ppn=28,mem=110GB
#PBS -j oe
#PBS -m bea
#PBS -A PAS1407
cd $HOME
module load R/3.5.0
R CMD BATCH --quiet --no-restore --no-save ./SubreadAlign.R > SubreadAlign.OUT" > script2
```

Then ```qsub script2```
